Begin adding notification_first_name to Simple Forms models

Thrillberg opened the pull request:
## Summary
This PR adds a method to every Simple Forms form model (including the base model), `notification_first_name`. This method is meant to specify the first name of the recipient of email notifications.

There is a bit of followup work from this:
1. Another PR to add methods for `notification_email_address`.
2. Modify our `NotificationEmail` to look at these methods instead of holding all the logic itself.

## Related issue(s)
https://github.com/orgs/department-of-veterans-affairs/projects/1443/views/3?sliceBy%5Bvalue%5D=Thrillberg&pane=issue&itemId=90296153&issue=department-of-veterans-affairs%7CVA.gov-team-forms%7C1908

pennja reviewed (dismissed):
Looks good, great work! 👍

Thrillberg commented:
Thanks @stevenjcumming ! I've corrected the typo (confirmed by @ksantiagoBAH on the team that owns this form to be a likely typo) and refactored as you recommended. I think since I still have to override Rubocop in some spots, it could use some more refactoring and it sounds like @ksantiagoBAH has that on his team's backlog now.

I'm seeing test failures that look unrelated. Do you think they are?

stevenjcumming commented:
> Thanks @stevenjcumming ! I've corrected the typo (confirmed by @ksantiagoBAH on the team that owns this form to be a likely typo) and refactored as you recommended. I think since I still have to override Rubocop in some spots, it could use some more refactoring and it sounds like @ksantiagoBAH has that on his team's backlog now.
> 
> I'm seeing test failures that look unrelated. Do you think they are?

It's related, there's a naming conflict, there's a class and module called `VBA4010007`. Not sure what I was thinking there. 

Perhaps you could rename the new class `SimpleFormsApi::Attachments::VBA4010007`?

Thrillberg commented:
> It's related, there's a naming conflict, there's a class and module called `VBA4010007`. Not sure what I was thinking there.
> 
> Perhaps you could rename the new class `SimpleFormsApi::Attachments::VBA4010007`? Or maybe there's a better name that I can't think of

Oh haha of course! I've just renamed it to `SimpleFormsApi::VBA4010007Attachment`. I think we'll want to refactor all of these models (they're all pretty big) into their own directories with a module for each. But that can come later!

Thrillberg commented:
@stevenjcumming This PR is ready for re-review! It's gotten quite huge though. Do you think I should break out the 40-10007 changes into a separate PR to merge first and then get to the main part of this one?

stevenjcumming commented:
> @stevenjcumming This PR is ready for re-review! It's gotten quite huge though. Do you think I should break out the 40-10007 changes into a separate PR to merge first and then get to the main part of this one?

@Thrillberg no it looks like it's just copy and paste move to another file. It does look much better though

Thrillberg commented on modules/simple_forms_api/app/models/simple_forms_api/base_form.rb:
```diff
@@ -12,5 +12,9 @@ def initialize(data)
     end
+
+    def notification_first_name
+      data.dig('veteran_full_name', 'first')
```
I think I'd rather change the others to be `data` (lose the `@`) since it's an attribute anyway. I'll update.
